applications/RomApplication/custom_utilities/rom_auxiliary_utilities.cpp

```diff
@@ -469,4 +469,36 @@ void RomAuxiliaryUtilities::GetPhiElemental(
         }
     }
 
+void RomAuxiliaryUtilities::GetPsiElemental(
+        Matrix &rPsiElemental,
+        const Element::DofsVectorType& rDofs,
+        const Element::GeometryType& rGeom,
+        const std::unordered_map<Kratos::VariableData::KeyType, Matrix::size_type>& rVarToRowMapping)
+    {
+        for(std::size_t i = 0; i < rDofs.size(); ++i)
+        {
+            const Dof<double>& r_dof = *rDofs[i];
+            if (r_dof.IsFixed())
+            {
+                noalias(row(rPsiElemental, i)) = ZeroVector(rPsiElemental.size2());
+            }
+            else
+            {
+                const auto it_node = std::find_if(rGeom.begin(), rGeom.end(),
+                    [&](const Node<3>& rNode)
+                    {
+                        return rNode.Id() == r_dof.Id();
+                    });
+                KRATOS_DEBUG_ERROR_IF(it_node == rGeom.end());
```

Comment:
Why only in debug? I'd always throw the error. Note that the code below will always crash if `it_node` equals the end iterator.
```suggestion
                KRATOS_ERROR_IF(it_node == rGeom.end());
```